mantidproject/mantid
# Implement slicing by log value

This PR adds a fourth slicing option to the event handling tab in the ISIS Reflectometry interface - log value slicing. This new slicing option allows the user to provide a selection of values (in a manner identical to custom slicing) and the name of a log. This allows a run to be filtered by the log values for each start/stop value, creating several slices.

~~NOTE: The `FilterByLogValue` algorithm currently has the property `TimeTolerance` set to `1.0` to avoid collecting no data when only one log is found. This is temporary however and we should talk to the instrument scientists about what they wish `TimeTolerance` to actually be set to.~~ As scientists are still undecided on its value, it will be left at a value of 1.0 for now.

**To test:**

This should be tested on `Release` mode. Alternatively if using `Debug` mode, go to `LoadEventNexus.cpp` and replace the following lines at lines 289-292 with this:

```
if (temp < 0) {
    if (temp < m_min_id)
        m_min_id = temp;
    if (temp > m_max_id)
        m_max_id = temp;
}
```

- Code review
- Access the interface from Interfaces->Reflectometry->ISIS Reflectometry
- Enter run `38415` with transmission run `38393` in the `Runs` tab
- In the `Event Handling` tab, ensure that omitting either slicing values or log name and processing causes the run to be processed as a non-event workspace (it will be processed via `GenericDataProcessorPresenter.cpp`).
- Enter values `0,100,1000,2000 and log name `raw_frames` and process. This should produce 3 sliced workspaces of each kind. As there are no positive-time `raw_frames` logs with values less than 100, plotting the first slice for IvsQ workspaces should show no data while the others will. Similarly, the first TOF slice should contain no data.
- You can check the log values for `raw_frames` by right clicking on the original TOF workspace, opening 'sample logs' and opening the logs for `raw_frames`.

Fixes #19012.

*Release notes have been updated.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
